PYME/recipes/base.py

```diff
@@ -918,37 +923,41 @@ def _inject_tables_from_hdf5(self, key, h5f, filename, extension):
                 from PYME.IO.ragged import RaggedVLArray
                 
                 rag = RaggedVLArray(h5f, t.name, copy=True) #force an in-memory copy so we can close the hdf file properly
-                rag.mdh = mdh
+                rag.mdh, rag.events = mdh, events
 
                 self.namespace[key_prefix + t.name] = rag
 
             elif isinstance(t, tables.table.Table):
                 #  pipe our table into h5r or hdf source depending on the extension
                 tab = tabular.H5RSource(h5f, t.name) if extension == '.h5r' else tabular.HDFSource(h5f, t.name)
-                tab.mdh = mdh
+                tab.mdh, tab.events = mdh, events
 
                 self.namespace[key_prefix + t.name] = tab
 
             elif isinstance(t, tables.EArray):
-                # load using ImageStack._loadh5, which finds metdata
-                im = ImageStack(filename=filename, haveGUI=False)
+                # load using ImageStack._loadh5
+                # FIXME - ._loadh5 will load events lazily, which isn't great if we got here after
+                # sending file over clusterIO inside of a context manager -> force it through since we already found it
+                im = ImageStack(filename=filename, mdh=mdh, events=events, haveGUI=False)
                 # assume image is the main table in the file and give it the named key
                 self.namespace[key] = im
 
     def loadInput(self, filename, key='input'):
         """
         Load input data from a file and inject into namespace
         """
+        #modify this to allow for different file types - currently only supports images
```

Comment:
```suggestion
```

patently untrue - supports tabular as either .h5r or .hdf. Raises an error on .csv and .xlsx.